src/packages/options/src/create.ts

```diff
@@ -22,40 +22,85 @@ export type Defaults<O extends Options> = {
   [K in keyof O]: Definitions<O[K]>;
 };
 
+const checkForConflicts = (
+  inputOptionName: string,
+  inputValue: unknown,
+  suppliedOptions: Map<string, { inputName: string; normalized: unknown }>,
+  conflicts?: string[]
+) => {
+  if (!conflicts) return;
+  for (const conflict of conflicts) {
+    if (suppliedOptions.has(conflict)) {
+      throw new Error(
+        `Values for both "${inputOptionName}" and ` +
+          `"${suppliedOptions.get(conflict).inputName}" cannot ` +
+          `be specified. They are mutually exclusive.`
+      );
+    }
+  }
+};
+
 function fill(defaults: any, options: any, target: any, namespace: any) {
   const def = defaults[namespace];
   const config = (target[namespace] = target[namespace] || {});
 
+  const suppliedOptions = new Map<
+    string,
+    { inputName: string; normalized: unknown }
+  >();
```

Comment:
Since we don't use the `normalized` value anywhere, we can simplify this to just 

```
const suppliedOptions = new Map<string, string>();
```

Reply:
Will we ever want to preserve what they input and the value? It could be that later on there could be conflicts related to certain values for different options? Or perhaps this is very premature...

Reply:
So anyways, if we change the format of the map will look like `<namespace>.<inputName> => value`? Or `<namespace>.<canonicalName> => <inputName>`? Hmmm...

Reply:
I don't think we need to store the `value` since we aren't going to do anything with it just yet. I'm a strong proponent of YAGNI (You Aren't Going to Need It)

Reply:
Yes, I'm also into the YAGNI